src/ConcreteOperations/intersection.jl

```diff
@@ -448,30 +417,14 @@ function intersection(X::Interval{N}, Y::LazySet{N}
         return EmptySet{N}(1)
     end
 end
-
 # symmetric method
-function intersection(Y::LazySet{N}, X::Interval{N}
-                     ) where {N<:Real}
-    return intersection(X, Y)
-end
+intersection(Y::LazySet, X::Interval) = intersection(X, Y)
 
-# disambiguation
-function intersection(X::Interval{N}, H::AbstractHyperrectangle{N}
-                     ) where {N<:Real}
-    return invoke(intersection, Tuple{Interval{N}, LazySet{N}}, X, H)
-end
-function intersection(H::AbstractHyperrectangle{N}, X::Interval{N}
-                     ) where {N<:Real}
-    return invoke(intersection, Tuple{Interval{N}, LazySet{N}}, X, H)
-end
-function intersection(X::Interval{N}, S::AbstractSingleton{N}
-                     ) where {N<:Real}
-    return invoke(intersection, Tuple{typeof(S), LazySet{N}}, S, X)
-end
-function intersection(S::AbstractSingleton{N}, X::Interval{N}
-                     ) where {N<:Real}
-    return invoke(intersection, Tuple{typeof(S), LazySet{N}}, S, X)
-end
+# disambiguations
+intersection(X::Interval, H::AbstractHyperrectangle) = _intersection_interval(X, H)
+intersection(H::AbstractHyperrectangle, X::Interval) = _intersection_interval(X, H)
+intersection(X::Interval, S::AbstractSingleton) = _intersection_interval(X, S)
+intersection(S::AbstractSingleton, X::Interval) = _intersection_interval(X, S)
```

Comment:
The old version used the `Singleton` method and I think this is is the more specific type.

Reply:
:+1:  i change this to `_intersection_singleton`